src/orion/benchmark/__init__.py

```diff
@@ -0,0 +1,330 @@
+        """
+        return self._id
+
+    @property
+    def configuration(self):
+        """Return a copy of an `Benchmark` configuration as a dictionary."""
+        config = dict()
+
+        config["name"] = self.name
+        config["algorithms"] = self.algorithms
+
+        targets = []
+        for target in self.targets:
+            str_target = {}
+            assessments = target["assess"]
+            str_assessments = []
+            for assessment in assessments:
+                str_assessments.append(assessment.configuration)
+            str_target["assess"] = str_assessments
+
+            tasks = target["task"]
+            str_tasks = []
+            for task in tasks:
+                str_tasks.append(task.configuration)
+            str_target["task"] = str_tasks
+        targets.append(str_target)
+        config["targets"] = targets
+
+        if self.id is not None:
+            config["_id"] = self.id
+
+        # print('config:', config)
+        return copy.deepcopy(config)
+
+
+class Study:
+    """
+    Benchmark definition
+
+    Parameters
+    ----------
+    benchmark: `Benchmark` instance
+    algorithms: list
+        Algorithms used for benchmark, each algorithm can be a string or dict.
+    assessment: list
+        `Assessment` instance
+    task: list
+        `Task` instance
+    """
+
+    def __init__(self, benchmark, algorithms, assessment, task):
+        self.algorithms = algorithms
+        self.assessment = assessment
+        self.task = task
+        self.benchmark = benchmark
+
+        self.assess_name = type(self.assessment).__name__
+        self.task_name = type(self.task).__name__
+        self.experiments_info = []
+
+    def setup_experiments(self):
+        """Setup experiments to run of the study"""
+        max_trials = self.task.max_trials
+        task_num = self.assessment.task_num
+        space = self.task.get_search_space()
+
+        for task_index in range(task_num):
+
+            for algo_index, algorithm in enumerate(self.algorithms):
+
+                experiment_name = (
+                    self.benchmark.name
+                    + "_"
+                    + self.assess_name
+                    + "_"
+                    + self.task_name
+                    + "_"
+                    + str(task_index)
+                    + "_"
+                    + str(algo_index)
+                )
+                experiment = create_experiment(
+                    experiment_name,
+                    space=space,
+                    algorithms=algorithm,
+                    max_trials=max_trials,
+                )
+                self.experiments_info.append((task_index, experiment))
+
+    def execute(self):
+        """Execute all the experiments of the study"""
+        blackbox_fun = self.task.get_blackbox_function()
+        max_trials = self.task.max_trials
+
+        for _, experiment in self.experiments_info:
+            # TODO: it is a blocking call
```

Comment:
I'm not sure it should be done differently. If the execution of the benchmark is to be consistent with the execution of the experiments (orion hunt), then scaling up should require running the benchmark in many workers in parallel.

Reply:
I was thinking about 2 consideration when I added this `TODO` comment.
1) Run the experiment in a non-blocking way, so in the same notebook, we can query the state of benchmark instead of waiting for all them done.
2) Another consideration is to support run multiple experiments in parallel, which is low priority at this moment I think. This is related to what you mentioned, instead of only one way to run the benchmark code multiple times, we can also achieve some level parallel capability with one one.